### ISSUE TITLE
Some questions about "lxc_templates".

### ISSUE THREAD

**Issue by soarteam** [maruos/blueprints#10]
![20180223144744](https://user-images.githubusercontent.com/11802769/36581578-94f59526-18a8-11e8-87ba-428e3e6a666c.png)

What are these changes?
Maru on the left and LXC-2 on the right.

**Comment by soarteam** [comment]
@pdsouza If I want to build my own image, I need to modify the template file?

**Comment by soarteam** [comment]
If I try to use the LXC-2 Debian template, I get the following problem.
But not with Maru.

root@Soar-PC# ./build.sh -b debian -n jessie64 -- -a arm64
--> loading distro plugin...
[ DEBIAN ] loading...
--> building image...
[ DEBIAN ] bootstrapping rootfs...
lxc-create: utils.c: get_template_path: 1400 No such file or directory - bad template: /home/Maru/vendor/maruos/blueprints/blueprint/debian/lxc/templates/debian.sh
lxc-create: lxccontainer.c: do_lxcapi_create: 1440 bad template: /home/Maru/vendor/maruos/blueprints/blueprint/debian/lxc/templates/debian.sh
lxc-create: tools/lxc_create.c: main: 318 Error creating container jessie64
--> cleaning up...

**Comment by soarteam** [comment]
However, I could build Debian successfully using any of the templates, but I did not test it on my phone because at the moment, Maru does not seem to be able to meet the everyday needs of the Nexus 5X.

**Comment by soarteam** [comment]
Obviously, armhf (32bit) is outdated, the vast majority of software requirements ARM64 or AMD64 (Metasploit,OpenVAS,etc.). So I'm not ready to test Maru for my Nexus5.

**Comment by soarteam** [comment]
Can I download pre-compiled Ubuntu? Let "build.sh" treatment?
http://cdimage.ubuntu.com/ubuntu-base/releases/16.04.1/release/ubuntu-base-16.04.2-base-arm64.tar.gz

**Comment by soarteam** [comment]
@pdsouza

**Comment by soarteam** [comment]
Enter the Ubuntu rootfs, you need:
Now in the rootfs directory.

apt-get install qemu-user-static
cp /usr/bin/qemu-aarch64-static  usr/bin
cp -b /etc/resolv.conf etc/

**Comment by soarteam** [comment]
First of all, do not allow Ubuntu to build ARM64 directly on AMD64.
Second, QEMU can not emulate urandom, so you can only use official sources.
Debian's software name contrast Ubuntu is still different.
Maru must complete all the preparatory work in order to install Maru's DEB, or APT will not be able to use.
But I have done a lot, the victory is in sight.

**Comment by soarteam** [comment]
I have built an incomplete Ubuntu16.04 ARM64.
Can not build SSH, can not build deb package "install", but has installed the package "Maru" and other things needed.
Logically, it can be lxc start.

**Comment by soarteam** [comment]
#!/bin/bash
# 
function mnt() {
    echo "MOUNTING"
    sudo mount -t proc /proc ${2}proc
    sudo mount -t sysfs /sys ${2}sys
    sudo mount -o bind /dev ${2}dev
    sudo mount -o bind /dev/pts ${2}dev/pts        
    sudo chroot ${2}
}
function umnt() {
    echo "UNMOUNTING"
    sudo umount ${2}proc
    sudo umount ${2}sys
    sudo umount ${2}dev/pts
    sudo umount ${2}dev
}
if [ "$1" == "-m" ] && [ -n "$2" ] ;
then
    mnt $1 $2
elif [ "$1" == "-u" ] && [ -n "$2" ];
then
    umnt $1 $2
else
    echo ""
    echo "Either 1'st, 2'nd or both parameters were missing"
    echo ""
    echo "1'st parameter can be one of these: -m(mount) OR -u(umount)"
    echo "2'nd parameter is the full path of rootfs directory(with trailing '/')"
    echo ""
    echo "For example: ch-mount -m /media/sdcard/"
    echo ""
    echo 1st parameter : ${1}
    echo 2nd parameter : ${2}
fi

**Comment by soarteam** [comment]
#!/bin/bash

function mnt() {
    echo "MOUNTING"
    sudo mount -t proc /proc ${2}proc
    sudo mount -t sysfs /sys ${2}sys
    sudo mount -o bind /dev ${2}dev
    sudo mount -o bind /dev/pts ${2}dev/pts        
    sudo chroot ${2}
}
function umnt() {
    echo "UNMOUNTING"
    sudo umount ${2}proc
    sudo umount ${2}sys
    sudo umount ${2}dev/pts
    sudo umount ${2}dev
}
if [ "$1" == "-m" ] && [ -n "$2" ] ;
then
    mnt $1 $2
elif [ "$1" == "-u" ] && [ -n "$2" ];
then
    umnt $1 $2
else
    echo ""
    echo "Either 1'st, 2'nd or both parameters were missing"
    echo ""
    echo "1'st parameter can be one of these: -m(mount) OR -u(umount)"
    echo "2'nd parameter is the full path of rootfs directory(with trailing '/')"
    echo ""
    echo "For example: ch-mount -m /media/sdcard/"
    echo ""
    echo 1st parameter : ${1}
    echo 2nd parameter : ${2}
fi

**Comment by soarteam** [comment]
root:x:0:
daemon:x:1:
bin:x:2:
sys:x:3:
adm:x:4:syslog
tty:x:5:
disk:x:6:
lp:x:7:
mail:x:8:
news:x:9:
uucp:x:10:
man:x:12:
proxy:x:13:
kmem:x:15:
dialout:x:20:
fax:x:21:
voice:x:22:
cdrom:x:24:
floppy:x:25:
tape:x:26:
sudo:x:27:
audio:x:29:
dip:x:30:
www-data:x:33:
backup:x:34:
operator:x:37:
list:x:38:
irc:x:39:
src:x:40:
gnats:x:41:
shadow:x:42:
utmp:x:43:
video:x:44:
sasl:x:45:
plugdev:x:46:
staff:x:50:
games:x:60:
users:x:100:
nogroup:x:65534:
systemd-journal:x:101:
systemd-timesync:x:102:
systemd-network:x:103:
systemd-resolve:x:104:
systemd-bus-proxy:x:105:
input:x:106:
crontab:x:107:
syslog:x:108:
netdev:x:109:
aid_system:x:1000:root
aid_radio:x:1001:root
aid_bluetooth:x:1002:root
aid_graphics:x:1003:root
aid_input:x:1004:root
aid_audio:x:1005:root
aid_camera:x:1006:root
aid_log:x:1007:root
aid_compass:x:1008:root
aid_mount:x:1009:root
aid_wifi:x:1010:root
aid_adb:x:1011:root
aid_install:x:1012:root
aid_media:x:1013:root
aid_dhcp:x:1014:root
aid_sdcard_rw:x:1015:root
aid_vpn:x:1016:root
aid_keystore:x:1017:root
aid_usb:x:1018:root
aid_drm:x:1019:root
aid_available:x:1020:root
aid_gps:x:1021:root
aid_media_rw:x:1023:root
aid_mtp:x:1024:root
aid_drmrpc:x:1026:root
aid_nfc:x:1027:root
aid_sdcard_r:x:1028:root
aid_clat:x:1029:root
aid_loop_radio:x:1030:root
aid_media_drm:x:1031:root
aid_package_info:x:1032:root
aid_sdcard_pics:x:1033:root
aid_sdcard_av:x:1034:root
aid_sdcard_all:x:1035:root
aid_logd:x:1036:root
aid_shared_relro:x:1037:root
aid_shell:x:2000:root
aid_cache:x:2001:root
aid_diag:x:2002:root
aid_net_bt_admin:x:3001:root
aid_net_bt:x:3002:root
aid_inet:x:3003:root
aid_net_raw:x:3004:root
aid_net_admin:x:3005:root
aid_net_bw_stats:x:3006:root
aid_net_bw_acct:x:3007:root
aid_net_bt_stack:x:3008:root
ssh:x:110:

**Comment by pdsouza** [comment]
@soarteam Have you tried building arm64 debian if you need 64-bit capabilities? Just use `./build.sh -b debian -n jessie64 -- -a arm64`

Yes, it appears that Ubuntu needs more work. Perhaps you can try a prebuilt image from one of our users who has it working on Maru? See https://groups.google.com/forum/#!topic/maru-os-dev/hY1oM930BkU.
